x-pack/plugins/spaces/public/share_saved_objects_to_space/components/relatives_footer.tsx

```diff
@@ -8,7 +8,7 @@
 import { EuiHorizontalRule, EuiText } from '@elastic/eui';
 import React, { useMemo } from 'react';
 
-import type { SavedObjectReferenceWithContext } from '@kbn/core/public';
+import type { SavedObjectReferenceWithContext } from '@kbn/core-saved-objects-api-server';
```

Comment:
__question:__ I see the issue description mentions that it's expected that `@kbn/core-saved-objects-api-browser depends on @kbn/core-saved-objects-api-server`, does it also apply to the type dependencies from the browser parts of other plugins to `@kbn/core-saved-objects-api-server` or is it just intermediate step and we'll eventually have `SavedObjectReferenceWithContext` and `SavedObjectsCollectMultiNamespaceReferencesResponse` exposed from either `@kbn/core-saved-objects-api-browser` or `@kbn/core-saved-objects-common`?

Reply:
Good question, to which we don't have a definitive answer atm. For now, the idea is that 'server' types (understand types coming from the server-side client/repository signatures) that are not directly used by publicly exposed Core public-side SO APIs are considered as 'server-side only' from Core standpoint, and therefor will not be exposed from public/common core packages. 

Note that, as explained in the issue's description, it's still fine to import such types coming from `-server` packages from the client side from browser-side plugins that uses it. 

Mid/long term, we'll probably want to introduce an additional layer of types for the param/response types of the SO **HTTP** APIs (which we currently don't have atm). Those types will be exposed from a common package, given they handle the client/server contract via these HTTP APIs.

Reply:
> Mid/long term, we'll probably want to introduce an additional layer of types for the param/response types of the SO HTTP APIs (which we currently don't have atm). Those types will be exposed from a common package, given they handle the client/server contract via these HTTP APIs.

Got it, it makes sense. Thank you for clarifying.